src/amchart/Pie.js

```diff
@@ -191,10 +191,75 @@
 
         this._chart.validateNow();
         this._chart.validateData();
+        console.log("width: " + this.width());
+        console.log("height: " +this.width());
     };
 
     Pie.prototype.render = function(callback) {
         return HTMLWidget.prototype.render.apply(this, arguments);
+   };
+
+    Pie.prototype.postUpdate = function (domNode, element) {
+         var context = this;
+         
+         var containerBoundingClientRect = context._element.select("svg").node().getBoundingClientRect(); // might need to change this selection (could get a bit more accurate results?)
+         this.d3LabelSelection = element.selectAll(".amcharts-pie-label");
+
+         var left = [];
+         var right = [];
+         var top = [];
+         var bottom = [];
+
+        this.d3LabelSelection.each(function(d, i) {
+            var labelTopPos = d3.select(this).node().getBoundingClientRect().top;
+            if (labelTopPos < containerBoundingClientRect.top) {
+                top.push(labelTopPos - containerBoundingClientRect.top);
+            }
+
+            var labelBottomPos = d3.select(this).node().getBoundingClientRect().bottom;
+            if (labelBottomPos > containerBoundingClientRect.bottom) {
+                bottom.push(containerBoundingClientRect.bottom - labelTopPos);
+
+            }
+
+            var labelRightPos = d3.select(this).node().getBoundingClientRect().right;
+            if (labelRightPos > containerBoundingClientRect.right) {
+                right.push(containerBoundingClientRect.right - labelRightPos);
+            }
+
+
+            var labelLeftPos = d3.select(this).node().getBoundingClientRect().left;
+            if (labelLeftPos < containerBoundingClientRect.left) {
+                left.push(labelLeftPos - containerBoundingClientRect.left);
+            }
+        });
+
+         var topOffset = top.length ? Math.min.apply(Math, top) : 0;
+         var bottomOffset = bottom.length ? Math.min.apply(Math, bottom) : 0;
+         var rightOffset = right.length ? Math.min.apply(Math, right) : 0;
+         var leftOffset = left.length ? Math.min.apply(Math, left) : 0;
```

Comment:
Is ```Math.min.apply(Math, left)``` a safe hack?  while arguments are Array "like" they are not quite the same thing...